### PR TITLE
Migrate some old controllers

### PR DESCRIPTION
**Pull request by Pierstoval**
- Move more Formcontrollers to generic
- Move maintenance exit to its own controller

### CODE REVIEW

**Comment by Pierstoval** [comment]
Done

File: src/Glpi/Config/LegacyConfigurators/StandardIncludes.php [hunk 1]
```diff
@@ -219,17 +220,7 @@ public function execute(): void
...
-                    echo __('Service is down for maintenance. It will be back shortly.');
-                    echo "\n";
-                } else {
-                    TemplateRenderer::getInstance()->display('maintenance.html.twig', [
-                        'title'            => "MAINTENANCE MODE",
-                        'maintenance_text' => $CFG_GLPI["maintenance_text"] ?? "",
-                    ]);
-                }
-                exit();
+                throw new MaintenanceException();
```
Comment (cedric-anne) [inline src/Glpi/Config/LegacyConfigurators/StandardIncludes.php:223]: Could you please move this in a dedicated PR ?

File: front/mailcollector.form.php [hunk 2]
```diff
@@ -1,107 +0,0 @@
...
-        "setup",
-        //TRANS: %s is the user login
-        sprintf(__('%s updates an item'), $_SESSION["glpiname"])
-    );
-    Html::back();
-} else if (isset($_POST["get_mails"])) {
-    $mailgate->check($_POST['id'], UPDATE);
-    $mailgate->collect($_POST["id"], 1);
-
-    Html::back();
```
Comment (cedric-anne) [inline front/mailcollector.form.php:103]: This action needs a specific controller. Please do it in a separate PR to streamline validation.

File: src/Glpi/Controller/GenericFormController.php [hunk 3]
```diff
@@ -196,7 +193,7 @@ public function displayForm(CommonDBTM $object, Request $request): Response
         }
 
         return $this->render('pages/generic_form.html.twig', [
-            'id' => $request->query->get('id', -1),
+            'id' => $request->query->get('id', 0),
```
Comment (cedric-anne) [inline src/Glpi/Controller/GenericFormController.php:196]: Could you explain this change? I am not sure that it will have no side effect.

Comment (Pierstoval) [reply src/Glpi/Controller/GenericFormController.php:196]: It is strange how "invalid IDs" are handled internally. If I let this value to `-1`, the route `/front/certificate.form.php`/`/Certificate/Form` will not display the QR code for the "Asset URL" field. With the value "0", it displays it. I tried to search and didn't find why, but if you have a clue, maybe this could be fixed downstream instead of in the generic controller 👌

Comment (trasher) [reply src/Glpi/Controller/GenericFormController.php:196]: Will be fixed by #18492

Comment (trasher) [reply src/Glpi/Controller/GenericFormController.php:196]: Since the `getId()` method will return `-1`, I guess this kind of change can indeed have impact, depending on how tests are done in the underlying code... We should use `isNewItem` rather than compare the value of `fields ['id']` or `getId()` but we currently cannot assume things are done right :/

Comment (Pierstoval) [reply src/Glpi/Controller/GenericFormController.php:196]: Great, will remove this change from my PR 👌

**Review by cedric-anne — APPROVED** [review]
I reverted the removal of scripts related to `CommonDBRelation` items. The user should probably not rbe edirected to the created item itself, but to the previous page (i.e. the tab that was used to add the relation).

**Comment by cedric-anne** [comment]
> I reverted the removal of scripts related to `CommonDBRelation` items. The user should probably not rbe edirected to the created item itself, but to the previous page (i.e. the tab that was used to add the relation).

@Pierstoval You can propose a solution applied only to `Appliance_Item_Relation` in a dedicated PR. If it works for this class, it will probably match most of the other `CommonDBRelation` classes.

**Review by trasher — APPROVED** [review]
